shell/server/core.js

```diff
@@ -140,17 +141,17 @@ const makeSandstormCore = (grainId) => {
   return new Capnp.Capability(new SandstormCoreImpl(grainId), SandstormCore);
 };
 
-class NotificationHandle {
-  constructor(notificationId, saved, persistentMethods) {
+class NotificationHandle extends PersistentImpl {
+  // TODO(cleanup): Move to a different file.
+
+  constructor(db, saveTemplate, notificationId) {
+    super(db, saveTemplate);
     this.notificationId = notificationId;
-    this.saved = saved;
-    _.extend(this, persistentMethods);
   }
 
   close() {
```

Comment:
> How does the code work at all if you don't convert the object to a capability?

node-capnp must be doing the conversion automatically, when the object is passed in as a return value to `addOngoing()`. Note that currently that's the only place that notification handles are constructed. They are never actually restored after being saved, as far as I can tell. Rather, the point of saving them is so that the supervisor can later call `SandstormCore.drop()` on them.

If I edit the notification handle registration to this:

```javascript
globalFrontendRefRegistry.register({
  frontendRefField: "notificationHandle",

  restore(db, saveTemplate, notificationId) {
    return new NotificationHandle(db, saveTemplate, notificationId);
  },
});
```

and fix the dangling `_this` reference below, then wakelock notifications immediately dismiss themselves!